R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
+
     # ===================================================
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
+
     tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Supply sector information for energy transformation sectors") %>%
+      add_units("unitless") %>%
+      add_comments("Written to all regions from A22.sector") %>%
       add_legacy_name("L222.Supplysector_en") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("energy/A22.sector", "common/GCAM_region_names") ->
       L222.Supplysector_en
 
     tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Subsector logit exponents of energy transformation sectors") %>%
+      add_units("unitless") %>%
+      add_comments("Written to all regions from A22.subsector_logit") %>%
       add_legacy_name("L222.SubsectorLogit_en") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("energy/A22.subsector_logit", "common/GCAM_region_names") ->
       L222.SubsectorLogit_en
 
+    if (exists("L222.SubsectorShrwt_en")) {
     tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Subsector shareweights of energy transformation sectors") %>%
+      add_units("unitless") %>%
+      add_comments("conditionally created from the subset of A22.subsector_shrwt with values in column 'year'.") %>%
+      add_comments("Does not exist by default") %>%
       add_legacy_name("L222.SubsectorShrwt_en") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
+      add_precursors("energy/A22.subsector_shrwt", "common/GCAM_region_names") %>%
       # typical flags, but there are others--see `constants.R`
       add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
       L222.SubsectorShrwt_en
+  } else {
+    tibble(x = NA) %>%
+      add_title("Data not created") %>%
+      add_units("Unitless") %>%
+      add_comments("Data not created") %>%
+      add_legacy_name("energy/L222.SubsectorShrwt_en") ->
+      L222.SubsectorShrwt_en
```

Comment:
Won't you need a `FLAG_NO_TEST` on these? ...hmm, no?

Reply:
It only makes them if they don't exist in the old system, so it would be testing against a non-existent file? I guess that means it passes okay.

Reply:
It may be worth noting that the if statements that evaluate to `TRUE` return a `debug at #2:` when you're using `browser()` but they don't throw it into debug if you're not already in debug mode--it appears to work fine when you run the driver and it passes all the tests. I've quintuple checked the syntax, so it's not a problem with that.